app/src/main/java/com/capstone/weatherapp/DetailsFragment.kt

```diff
@@ -55,13 +55,44 @@ class DetailsFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         viewModel.city.observe(viewLifecycleOwner) {
-            updateBinding(binding, it)
+            when(it) {
+                is State.Success -> showSuccessState(it.result)
+                is State.Loading -> showLoadingState()
+                is State.Error -> showErrorState(it.message)
+            }
         }
         viewModel.getSingleCityFromRepo(cityId)
     }
 
+    fun showLoadingState() {
+        binding.detailsLayout.visibility = View.GONE
+        binding.stateLayout.visibility = View.VISIBLE
+        binding.detailsState.text = ""
+        Glide.with(this)
```

Comment:
same as above comment regarding Glide, should just need to set the visibility of this element instead of having to set the image source with Glide every time